# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1225372512 -3600
#      Thu Oct 30 14:15:12 2008 +0100
# Node ID 53f35799b2355409f6b3883069cfab2136127765
# Parent  7cbe01c21986a589fdc84baf4e8840b0c5f1f9be
Add support for left/right argument to the legend function

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,14 @@
 2008-10-30  David Bateman  <dbateman@free.fr>
 
+	* plot/legend.m: Add support for the "left" and "right" options.
+	* plot/__go_draw_axes__.m: If the axes property keyreverse is set,
+	reverse the key and labelling text.
+	
 	* plot/__clabel__.m, plot/clabel.m: New functions.
 	* plot/Makefile.in (SOURCES): Add them here.
 	* plot/__contour__.m: Rewrite to use contour groups.
 	* plot/contourf.m: Call __contour__ instead of using specific code.
 	* plot/contour.m, plot/contour3.m: Minor modification to allow for
 	new interface to __contour__.
 
 2008-10-29  Thorsten Meyer  <thorsten.meyier@gmx.de>
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1048,16 +1048,21 @@ function __go_draw_axes__ (h, plot_strea
     endif
 
     if (strcmpi (axis_obj.key, "on"))
       if (strcmpi (axis_obj.keybox, "on"))
 	box = "box";
       else
 	box = "nobox";
       endif
+      if (strcmpi (axis_obj.keyreverse, "on"))
+	reverse = "reverse";
+      else
+	reverse = "noreverse";
+      endif
       inout = "inside";
       keypos = axis_obj.keypos;
       if (ischar (keypos))
 	keypos = lower (keypos);
 	keyout = findstr (keypos, "outside");
 	if (! isempty (keyout))
 	  inout = "outside";
 	  keypos = keypos(1:keyout-1);
@@ -1094,17 +1099,17 @@ function __go_draw_axes__ (h, plot_strea
 	case "best" 
 	  pos = "";
 	  warning ("legend: 'Best' not yet implemented for location specifier.\n");
 	  ## least conflict with data in plot
 	  ## least unused space outside plot
 	otherwise
 	  pos = "";
       endswitch
-      fprintf (plot_stream, "set key %s %s %s;\n", inout, pos, box);
+      fprintf (plot_stream, "set key %s %s %s %s;\n", inout, pos, box, reverse);
     else
       fputs (plot_stream, "unset key;\n");
     endif
 
     fputs (plot_stream, "set style data lines;\n");
 
     if (! use_gnuplot_for_images)
       for i = 1:ximg_data_idx
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -125,16 +125,20 @@ function legend (varargin)
 	      set (ca, "key", "off");
 	    else
 	      set (ca, "key", "on");
 	    endif
 	  case "boxon"
 	    set (ca, "key", "on", "keybox", "on");
 	  case "boxoff"
 	    set (ca, "keybox", "off");
+	  case "left"
+	    set (ca, "keyreverse", "off")
+	  case "right"
+	    set (ca, "keyreverse", "on")
 	  otherwise
 	    typ = get (kids (k), "type");
 	    while (k <= nkids && ! strcmp (typ, "line") &&
 		   ! strcmp (typ, "hggroup"))
 	      k++;
 	      typ = get (kids (k), "type");
 	    endwhile
 	    if (k <= nkids)
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2008-10-30  David Bateman  <dbateman@free.fr>
+
+	* graphic.h.in (axes::properties): Add keyreverse property.
+	* graphic.cc (axes::properties::set_defaults): Initialize
+	keyreverse property.
+
 2008-10-28  Jaroslav Hajek <highegg@gmail.com>
 
 	* Cell.h (Cell::assign (const Array<idx_vector>&, ...),
 	Cell::delete_elements (const Array<idx_vector>&, ...)):
 	New member functions.
 	* Cell.cc (Cell::assign (const octave_value_list&, ...),
 	Cell::delete_elements (const octave_value_list&, ...)):
 	Call Array<T>::assign.
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -2288,16 +2288,17 @@ axes::properties::set_title (const octav
 
 void
 axes::properties::set_defaults (base_graphics_object& obj,
 				const std::string& mode)
 {
   box = "on";
   key = "off";
   keybox = "off";
+  keyreverse = "off";
   keypos = 1.0;
   colororder = default_colororder ();
   dataaspectratio = Matrix (1, 3, 1.0);
   dataaspectratiomode = "auto";
   layer = "bottom";
 
   Matrix tlim (1, 2, 0.0);
   tlim(1) = 1;
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -2554,16 +2554,17 @@ public:
 
     // properties which are not in matlab: interpreter
 
     BEGIN_PROPERTIES (axes)
       array_property position u , default_axes_position ()
       bool_property box , "on"
       bool_property key , "off"
       bool_property keybox , "off"
+      bool_property keyreverse , "off"
       any_property keypos , 1
       array_property colororder , default_colororder ()
       array_property dataaspectratio m , Matrix (1, 3, 1.0)
       radio_property dataaspectratiomode , "{auto}|manual"
       radio_property layer , "{bottom}|top"
       row_vector_property xlim mu , default_lim ()
       row_vector_property ylim mu , default_lim ()
       row_vector_property zlim mu , default_lim ()
